packages/v-protocol/src/parser.ts

```diff
@@ -239,28 +244,51 @@ export class Parser {
     return new CommandCompleteMessage(length, text)
   }
 
-  private parseCopyData(offset: number, length: number, bytes: Buffer) {
-    const chunk = bytes.slice(offset, offset + (length - 4))
-    return new CopyDataMessage(length, chunk)
+  private parseVerifyFilesMessage(offset: number, length: number, bytes: Buffer) {
+    this.reader.setBuffer(offset, bytes)
+    const numFiles = this.reader.int16() //int16 number of files, n
+    const fileNames: string[] = new Array(numFiles)
+    for (let i = 0; i < numFiles; i++) {
+      fileNames[i] = this.reader.cstring() //string[n], name of each file
+    }
+    const rejectFile = this.reader.cstring() //string reject file name
+    const exceptionFile = this.reader.cstring() //string exceptions file name
+    return new VerifyFilesMessage(length, numFiles, fileNames, rejectFile, exceptionFile)
   }
 
-  private parseCopyInMessage(offset: number, length: number, bytes: Buffer) {
-    return this.parseCopyMessage(offset, length, bytes, 'copyInResponse')
+  private parseWriteFileMessage(offset: number, length: number, bytes: Buffer) {
+    this.reader.setBuffer(offset, bytes)
+    const fileName = this.reader.cstring()
+    const fileLength = this.reader.int32()
+    let fileContents: string | bigint[]
+    // if filename is empty, it means we used returnrejected instead of rejection file, the fileLength 
+    // will be in mutliples of 8 bytes for each rejected row number in Little Endian 64 bit format
+    if (fileName.length === 0) {
+      fileContents = []
+      for (let i = 0; i < fileLength; i += 8) {
+          fileContents.push(this.reader.int64LE())
+      }
+    } else {
+      fileContents = this.reader.string(fileLength)
```

Comment:
I will modify to read file contents in chunks for safety

Reply:
So it looks like the server already sends Writefile messages containing rejects/exceptions in chunks of 8Kb. If it's larger than 8Kb there are multiple Writefile messages sent in a row. That seems small enough to me, only 8Kb needs to be in memory at a time before writing to disk

Reply:
Is the python client reading chunks smaller than 8kb? Is that necessary? 

We might want to update the protocol doc with that information, as it was a surprise to me too. I only discovered it by using the dc_client_server_messages table and seeing a string of WriteFile messages from the backend in a row. Each 8Kb in size except for the last which was just contained the remaining data

Reply:
That would be great. I did find this line in the guide: 

"The backend should then send zero or more WriteFile messages for rejected row descriptions and an EndOfBatchResponse message for acknowledgement of data loading from this file."

So there is a small hint that there could be multiple messages but we probably could elaborate that, and possibly the diagram as well if you wanted to.

Reply:
We modify contents but not the mechanism for message transfer to the frontend when extend_copy_reject_info=True, so I expect there is an 8kb limit. I would operate under that assumption, and I will test and verify this for the protocol doc

Reply:
In case we have not discussed this elsewhere ( I can't remember anymore ) I have verified that the extend_copy_reject_info property has no impact on the size of the writefile message, it is still a max of 8Kb